examples/common/execution.py

```diff
@@ -30,6 +31,12 @@ class ExecutionMode:
     GPU_DATAPARALLEL = "gpu_dataparallel"
 
 
+class ExecutionParameters:
+    def __init__(self, cpu_only: bool, current_gpu):
+        self.cpu_only = cpu_only
+        self.current_gpu = current_gpu
```

Comment:
Since this goes into an API function, this struct itself should also go into the API files. Also, add a type hint for `current_gpu`.

Reply:
Done